WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt

```diff
@@ -385,22 +394,57 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
                             this.siteId,
                             this.feedId
                     )
+                    is ReaderNavigationEvents.SharePost -> ReaderActivityLauncher.sharePost(context, post)
+                    is ReaderNavigationEvents.OpenPost -> ReaderActivityLauncher.openPost(context, post)
+                    is ReaderNavigationEvents.ShowReportPost -> {
+                        ReaderActivityLauncher.openUrl(
+                                context,
+                                readerUtilsWrapper.getReportPostUrl(url),
+                                OpenUrlType.INTERNAL
+                        )
+                    }
                     is ReaderNavigationEvents.ShowReaderComments,
                     is ReaderNavigationEvents.ShowNoSitesToReblog,
                     is ReaderNavigationEvents.ShowSitePickerForResult,
                     is ReaderNavigationEvents.OpenEditorForReblog,
                     is ReaderNavigationEvents.ShowBookmarkedTab,
-                    is ReaderNavigationEvents.ShowBookmarkedSavedOnlyLocallyDialog,
-                    is ReaderNavigationEvents.SharePost,
-                    is ReaderNavigationEvents.OpenPost,
-                    is ReaderNavigationEvents.ShowReportPost -> { // TODO: Handle nav events
+                    is ReaderNavigationEvents.ShowBookmarkedSavedOnlyLocallyDialog -> { // TODO: Handle nav events
                     }
                 }
             }
         })
         viewModel.start()
     }
 
+    private fun updateMoreMenu(
+        state: ReaderPostDetailsUiState
+    ) {
+        val moreMenu: View? = toolBar.findViewById(R.id.menu_more)
+        moreMenu?.let {
+            it.setOnClickListener { state.onMoreButtonClicked.invoke(state) }
+            state.moreMenuItems?.let {
+                renderMoreMenu(state, state.moreMenuItems, moreMenu)
+            }
+        }
+    }
+
+    private fun renderMoreMenu(uiState: ReaderPostDetailsUiState, actions: List<SecondaryAction>, v: View) {
+        // TODO: Add Tracks
+        val listPopup = ListPopupWindow(v.context)
```

Comment:
I'm not sure this will work as intended and it might be a bit confusing to get oriented in. It seems the VM is responsible for showing the more menu - by updating the UIState. However, the view layer is responsible for dismissing the menu - AFAICT no-one updates the UIState. We might want to consider moving the dismiss logic into the VM layer as well. Wdyt?

Reply:
I tested the app and found a semi-related crash. The app crashes on rotation.
1. Open the more menu
2. Rotate your device

```
020-10-14 10:40:06.648 24742-24742/org.wordpress.android.beta E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.wordpress.android.beta, PID: 24742
    android.view.WindowManager$BadTokenException: Unable to add window -- token null is not valid; is your activity running?
        at android.view.ViewRootImpl.setView(ViewRootImpl.java:907)
        at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:387)
        at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:95)
        at android.widget.PopupWindow.invokePopup(PopupWindow.java:1577)
        at android.widget.PopupWindow.showAsDropDown(PopupWindow.java:1424)
        at androidx.appcompat.widget.AppCompatPopupWindow.showAsDropDown(AppCompatPopupWindow.java:77)
        at androidx.core.widget.PopupWindowCompat.showAsDropDown(PopupWindowCompat.java:69)
        at androidx.appcompat.widget.ListPopupWindow.show(ListPopupWindow.java:755)
        at org.wordpress.android.ui.reader.ReaderPostDetailFragment.renderMoreMenu(ReaderPostDetailFragment.kt:445)
        at org.wordpress.android.ui.reader.ReaderPostDetailFragment.updateMoreMenu(ReaderPostDetailFragment.kt:426)
        at org.wordpress.android.ui.reader.ReaderPostDetailFragment.access$updateMoreMenu(ReaderPostDetailFragment.kt:142)
        at org.wordpress.android.ui.reader.ReaderPostDetailFragment$initViewModel$1.onChanged(ReaderPostDetailFragment.kt:375)
        at org.wordpress.android.ui.reader.ReaderPostDetailFragment$initViewModel$1.onChanged(ReaderPostDetailFragment.kt:142)
```

I think we'll need to store a reference to the PopupWindow and dismiss it in "onDestroy". I'd treat it as a dialog. Wdyt?

Reply:
Fixed in 915ad23